app/controllers/CsvFileUploadController.scala

```diff
@@ -176,6 +179,29 @@ class CsvFileUploadController @Inject()(val mcc: MessagesControllerComponents,
     }
   }
 
+  def checkFileNames(csvCallbackData: List[UploadedSuccessfully], schemeInfo: SchemeInfo)(implicit authContext: ERSAuthData,
+                                                                                          request: Request[AnyRef],
+                                                                                          hc: HeaderCarrier): Future[Result] = {
+    ersUtil.fetch[UpscanCsvFilesList](ersUtil.CSV_FILES_UPLOAD, schemeInfo.schemeRef).flatMap { list =>
+      val uploadedWithCorrectName: Boolean = list.ids
+        .map(expectedFile => expectedFile.fileId)
+        .zip(csvCallbackData.reverse
+          .map(uploadedFile => uploadedFile.name))
+        .map(t => (ersUtil.getPageElement(schemeInfo.schemeId, ersUtil.PAGE_CHECK_CSV_FILE, t._1 + ".file_name"), t._2))
+        .forall(names => names._1 == names._2)
+      if (uploadedWithCorrectName) {
+        validateCsv(csvCallbackData, schemeInfo)
+      } else {
+        Logger.warn(s"[CsvFileUploadController][checkFileNames] User uploaded the wrong file")
```

Comment:
does this need to be warn-level?

Reply:
Happy for it to be info, tried to keep the levels in-line with what I was seeing in the rest of the service.